controllers/hetznercluster_controller.go

```diff
@@ -194,7 +193,9 @@ func (r *HetznerClusterReconciler) reconcileNormal(ctx context.Context, clusterS
 	}
 	conditions.MarkTrue(hetznerCluster, infrav1.PlacementGroupsSynced)
 
-	if hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled {
+	if !hetznerCluster.Spec.ControlPlaneLoadBalancer.Enabled && (hetznerCluster.Spec.ControlPlaneEndpoint == nil || len(hetznerCluster.Spec.ControlPlaneEndpoint.Host) == 0) {
```

Comment:
nit: `len(hetznerCluster.Spec.ControlPlaneEndpoint.Host) == 0` since Host is a string, I would prefer `... == ""`. Or do you have special reasons to check for the length?

Reply:
I'm just used to checking the length of strings, I'll cope with your coding style!